### PR TITLE
start is now added to the path

### PR DESCRIPTION
**Pull request by jwallace42**
---

## Basic Info

| Info | Please fill out this column |
| ------ | ----------- |
| Ticket(s) this addresses   | #3085 |
| Primary OS tested on | Ubuntu |
| Robotic platform tested on | Gazebo |

---

## Description of contribution in a few bullet points
Start point is now added to the path

## Description of documentation updates required from your changes

---

#### For Maintainers: 
- [ ] Check that any new parameters added are updated in navigation.ros.org
- [ ] Check that any significant change is added to the migration guide
- [ ] Check that any new features **OR** changes to existing behaviors are reflected in the tuning guide
- [ ] Check that any new functions have Doxygen added
- [ ] Check that any new features have test coverage
- [ ] Check that any new plugins is added to the plugins page
- [ ] If BT Node, Additionally: add to BT's XML index of nodes for groot, BT package's readme table, and BT library lists

### CODE REVIEW

File: nav2_smac_planner/src/node_2d.cpp
Comment (SteveMacenski) [inline nav2_smac_planner/src/node_2d.cpp:162]: Will have to wait for CI, but I think the while should be on the same line as the end bracket by styling, but LGTM if linting is happy!

**Review by SteveMacenski — APPROVED** [review]

**Comment by mangoschorle** [comment]
@SteveMacenski  @jwallace42 

This is still incorrect
You need a final `path.push_back(
      Node2D::getCoords(current_node->getIndex()));` after the do while loop

**Comment by jwallace42** [comment]
https://github.com/ros-planning/navigation2/pull/3168

Whoops, 

here is the fix. For some reason I only though we needed the do while loop to solve the issue. 

Thanks for pointing that out. :)

**Pull request merged**
